### PR TITLE
Remove block processing channel

### PR DESCRIPTION
**Pull request by batconjurer**
The tests pass at least. And there isn't that unbounded channel.

### CODE REVIEW

**Comment by sug0** [comment]
We should pin this in the namada hall of fame

**Review by sug0 — APPROVED** [review]

File: apps/src/lib/node/ledger/ethereum_node/oracle.rs [hunk 1]
```diff
@@ -318,8 +306,8 @@ async fn process(
                 pending = pending.len(),
                 confirmed = confirmed.len(),
                 ?MIN_CONFIRMATIONS,
-                "Some events have reached the minimum number of confirmations \
-                 and will be sent onwards"
+                "Some events that have reached the minimum number of \
+                 confirmations and will be sent onwards"
```
Comment (james-chf) [inline apps/src/lib/node/ledger/ethereum_node/oracle.rs:310]: The typo here was fixed already in the other PR

File: apps/src/lib/node/ledger/ethereum_node/test_tools/mod.rs [hunk 2]
```diff
@@ -206,6 +215,13 @@ pub mod mock_web3_client {
                         client.events.push((event_ty, data, height, seen));
                     }
                 }
+                if client.last_block_processed < Some(block_to_check.clone()) {
```
Comment (sug0) [inline apps/src/lib/node/ledger/ethereum_node/test_tools/mod.rs:218]:
```suggestion
                if client.last_block_processed.as_ref() < Some(&block_to_check) {
```
I think we might even be able to compare by ref here

**Comment by james-chf** [comment]
This PR looks like it would change it so that we're no longer directly testing the oracle fully processing block heights. Instead we test indirectly that it is calling the fake `Web3Client::check_for_events` at least once for each expected block height. Why not have the oracle directly make observable what it is doing, and test on that?

**Comment by batconjurer** [comment]
> This PR looks like it would change it so that we're no longer directly testing the oracle fully processing block heights. Instead we test indirectly that it is calling the fake `Web3Client::check_for_events` at least once for each expected block height. Why not have the oracle directly make observable what it is doing, and test on that?

I would love it if we could test directly, but that seems to involve invasive code changes and test artifacts in production code flows, which I really don't like. I was just trying to come up with an alternate solution using an existing mock. Otherwise, the best option is probably to make an oracle with generics in order to make a mock version for tests.

**Review by james-chf — APPROVED** [review]
Sure, I think adding code just for test paths is not great, that said more generally I think it would be good to make our components observable for production, by exposing things that could be published as a metric (https://github.com/anoma/namada/issues/27) - then that same instrumentation for observability can be used in tests. I think we should definitely revisit this before launch.

Feel free to take over the other PR to merge this and then that, though this isn't the approach I would have taken.

Re. making an oracle with generics for tests (I'm guessing making the oracle generic over it's `Web3Client`?), I think that would be useful, I had a go at this a while back but wasn't able to do it because of the `web30` thread issue (https://github.com/althea-net/web30/issues/4). Specifically trying to implement an `[async_trait]` `Web3Client` was difficult.

File: apps/src/lib/node/ledger/ethereum_node/oracle.rs [hunk 3]
```diff
@@ -439,6 +424,7 @@ mod test_oracle {
             oracle,
             mut eth_recv,
             admin_channel,
+            blocks_processed_recv: _processed,
```
[the remark this answers is not in the document]
Comment (batconjurer) [reply apps/src/lib/node/ledger/ethereum_node/oracle.rs:427]: If we leave it out, Rust eager drops it and causes a panic on the channel send method in the mock oracle.

File: apps/src/lib/node/ledger/ethereum_node/test_tools/mod.rs (hunk 2 above)
Comment (sug0) [reply apps/src/lib/node/ledger/ethereum_node/test_tools/mod.rs]: yeah, jacob was also surprised about this lol. it's definitely a bit of an obscure-ish feature of `Option` types, I think.

**Pull request merged**